msm/src/prover.rs

```diff
@@ -105,21 +169,100 @@ where
                     .map(eval)
                     .collect::<Vec<_>>();
                 let sum = eval(&lookup_env.lookup_aggregation_poly_d1);
-                LookupProof { m, h, sum }
+                (m, h, sum)
             };
-            (Some(evals(&zeta)), Some(evals(&zeta_omega)))
+            let (m_zeta, h_zeta, sum_zeta) = evals(&zeta);
+            let (m_zeta_omega, h_zeta_omega, sum_zeta_omega) = evals(&zeta_omega);
+            Some(LookupProof {
+                m: m_zeta
+                    .into_iter()
+                    .zip(m_zeta_omega.into_iter())
+                    .map(|(zeta, zeta_omega)| PointEvaluations { zeta, zeta_omega })
+                    .collect(),
+                h: h_zeta
+                    .into_iter()
+                    .zip(h_zeta_omega.into_iter())
+                    .map(|(zeta, zeta_omega)| PointEvaluations { zeta, zeta_omega })
+                    .collect(),
+                sum: PointEvaluations {
+                    zeta: sum_zeta,
+                    zeta_omega: sum_zeta_omega,
+                },
+            })
         } else {
-            (None, None)
+            None
         }
     };
-    // -- Start opening proof - Preparing the Rust structures
-    let group_map = G::Map::setup();
 
-    // Gathering all polynomials to use in the opening proof
-    let polynomials: Vec<_> = polys.into_iter().collect();
+    let witness_evals = {
+        let WitnessColumns { x } = &witness_polys;
+        let eval_foo = |point| move |poly: &DensePolynomial<G::ScalarField>| poly.evaluate(point);
+        WitnessColumns {
+            x: x.iter()
+                .map(eval_foo(&zeta))
+                .zip(x.iter().map(eval_foo(&zeta_omega)))
+                .map(|(zeta, zeta_omega)| PointEvaluations { zeta, zeta_omega })
+                .collect::<Vec<_>>(),
+        }
+    };
+
+    let proof_evals: ProofEvaluations<PointEvaluations<G::ScalarField>> = {
+        ProofEvaluations {
+            public_evals: None,
+            witness_evals,
+            mvlookup_evals,
+        }
+    };
 
-    let mut polynomials: Vec<_> = polynomials
-        .iter()
+    // Challenges and constants used for evaluating expressions
+    let expr_chals = Challenges {
```

Comment:
Let's make a small PR adding these, built on top of https://github.com/o1-labs/proof-systems/pull/1850